### PR TITLE
tests: skip the test-snapd-timedate-control-consumer.date to avoid NTP sync error

### PR DESCRIPTION
**Pull request by sergiocazzolato**
NTP syncronization is not done after the date is changed using the date
command. The issue happens even when the syncronization is manually forced.
The error displayed in the journal is the following:
synchronized.go:72: DEBUG: NTPSynchronized state returned by timedate1:
false

In order to reproduce the error and check the fix, run:
spread-debug -order -single-worker -debug
google:ubuntu-20.04-64:tests/main/interfaces-time-control
google:ubuntu-20.04-64:tests/main/snapd-stat

### CODE REVIEW

File: tests/main/interfaces-time-control/task.yaml [hunk 1]
```diff
@@ -53,8 +56,14 @@ execute: |
     [ "$(test-snapd-timedate-control-consumer.timedatectl-time status | grep -oP 'RTC in local TZ: \K(.*)')" = "no" ]
 
     echo "And direct setting via date"
-    nice_when="$(date -d 'tomorrow' +'%m%d%H%M.%y')"
-    test-snapd-timedate-control-consumer.date "$nice_when"
+    # Ntp syncronization is not done after the date is changed using the date command.
```
Comment (bboozzoo) [inline tests/main/interfaces-time-control/task.yaml:59]: linter complains about these:
```
tests/main/interfaces-time-control/task.yaml:59:10: "syncronization" is a misspelling of "synchronization"
tests/main/interfaces-time-control/task.yaml:60:38: "syncronization" is a misspelling of "synchronization"
```

File: tests/main/interfaces-time-control/task.yaml [hunk 2]
```diff
@@ -16,11 +16,14 @@ prepare: |
...
+    snap connect test-snapd-timedate-control-consumer:netlink-audit
+    snap connect test-snapd-timedate-control-consumer:time-control
     if [ -f rtc.txt ]; then
-        timedatectl set-local-rtc "$(cat rtc.txt)"
+        test-snapd-timedate-control-consumer.timedatectl-time set-local-rtc "$(cat rtc.txt)"
     fi
 
-    date "$(cat now.txt)"
+    # TODO: re-enable once the ntp issue is fixed
+    #test-snapd-timedate-control-consumer.date "$(cat now.txt)"
```
Comment (bboozzoo) [inline tests/main/interfaces-time-control/task.yaml:26]: so can we use timedatectl set-time instead then?

File: tests/main/interfaces-time-control/task.yaml [hunk 3]
```diff
@@ -16,11 +16,14 @@ prepare: |
 
 restore: |
     # Restore the initial rtc configuration
+    snap connect test-snapd-timedate-control-consumer:netlink-audit
+    snap connect test-snapd-timedate-control-consumer:time-control
     if [ -f rtc.txt ]; then
-        timedatectl set-local-rtc "$(cat rtc.txt)"
+        test-snapd-timedate-control-consumer.timedatectl-time set-local-rtc "$(cat rtc.txt)"
```
Comment (bboozzoo) [inline tests/main/interfaces-time-control/task.yaml:22]: I'm quite confused why we're using the snap we did the testing with here but not the host's timedatectl?

Comment (sergiocazzolato) [reply tests/main/interfaces-time-control/task.yaml:22]: I'll revert that, it was just for testing

**Review by mvo5 — COMMENTED** [review]
Thanks for getting to the bottom of this. one suggestion inline.

**Comment by mvo5** [comment]
I was wondering if https://github.com/snapcore/snapd/pull/11067 might work ? With that we won't need to disable the tests, it just restores the ntp setting - I did not run your spread-debug to validate, if @sergiocazzolato  could quickly check that would be amazing!

File: tests/main/interfaces-time-control/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-time-control/task.yaml]: I tried that, but when the problem comes when the test does:
test-snapd-timedate-control-consumer.date "$nice_when"

File: tests/main/interfaces-time-control/task.yaml [hunk 4]
```diff
@@ -16,11 +16,14 @@ prepare: |
 
 restore: |
     # Restore the initial rtc configuration
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/main/interfaces-time-control/task.yaml]: I already tried that, it was 1 of the first things I tried, but the problem is still there if we restore the initial ntp value, seems to be an issue related to changing the date. I see that ntp is enabled but the sync is not done and the refresh still fail.

[the remark this answers is not in the document]
Comment (mvo5) [reply tests/main/interfaces-time-control/task.yaml]: Thanks! I think we should merge this PR and open one in parallel that reverts it and tries to come up with a fix. maybe we just need to run "timedatectl set-ntp true" and then wait for it to have synced or something :/

**Review by mvo5 — APPROVED** [review]
Let's merge this so that we get green tests again and open in parallel a revert where we explore how to fix the issue for good.

File: tests/main/interfaces-time-control/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-time-control/task.yaml]: updated

File: tests/main/interfaces-time-control/task.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply tests/main/interfaces-time-control/task.yaml]: yes, I already tried that yesterday and after a time (list 90s) the sync is not done and the refresh fails. Not sure how long does it take to be on sync again

**Comment by mvo5** [comment]
I merged it now, we will need some sort of followup that reverts, fixes it but it's not urgent, the "flaky-tests" label will help us to remember this :)

**Pull request merged**